server/auth/registration.go

```diff
@@ -3,39 +3,28 @@ package auth
 import (
 	"github.com/8tomat8/SSU-Golang-252-Chat/database"
 	"github.com/8tomat8/SSU-Golang-252-Chat/loger"
+	"github.com/8tomat8/SSU-Golang-252-Chat/messageService"
 	_ "github.com/jinzhu/gorm/dialects/sqlite"
 )
 
-//type UserRegistry struct {
-//	UserName string `json:"user_name"`
-//	Password string `json:"password"`
-//	NickName string `json:"nick_name"`
-//}
-
-type User struct {
-	UserName string `gorm:"primary_key"`
-	Password string
-	NickName string
-}
-
-func NewUser(user *User, UserName, Password string) error {
+func RegisterNewUser(user *messageService.User) (*messageService.User, string, error) {
 	db, err := database.GetStorage()
 	if err != nil {
 		loger.Log.Errorf("Failed to open db", err)
-		return err
+		return nil, "", err
 	}
-	if err := db.Where("username = ?", UserName).First(&user).Error; err != nil {
+	if err := db.Where("username = ?", user.UserName).Error; err != nil {
 		loger.Log.Errorf("no such user in db", err)
```

Comment:
[Read this article](http://jinzhu.me/gorm/advanced.html). There is good example for this case.